MiniApp/Classes/Protocols/MiniAppMessageDelegate.swift

```diff
@@ -14,11 +14,45 @@ public protocol MiniAppMessageDelegate: MiniAppUserInfoDelegate, MiniAppShareCon
     /// result (allowed/denied) should be returned.
     func requestPermission(permissionType: MiniAppPermissionType, completionHandler: @escaping (Result<MASDKPermissionResponse, MASDKPermissionError>) -> Void)
 
+    /// Optional Interface that can be implemented in the host app to handle the Custom Permissions.
+    /// SDK will open its default UI for prompting Custom Permissions request if this protocol interface is not overridden
+    func requestCustomPermissions(permissions: [MASDKCustomPermissionModel],
+                                  miniAppTitle: String,
+                                  completionHandler: @escaping (Result<[MASDKCustomPermissionModel], MASDKCustomPermissionError>) -> Void)
+
     /// Interface that should be implemented in the host app to handle the Custom Permissions.
     /// Host app is responsible to display the alert/dialog with the [MiniAppCustomPermissionType] permissions to the user and the result should be returned back to the SDK
+    @available(*, deprecated,
+    message:"Since version 2.3.0, you can requestCustomPermissions using the below method",
```

Comment:
"the below method" makes no sense when autocompleting in Xcode. the renamed section gives enough informations about the method to switch to

Reply:
I happen to remove this method now. I was thinking about whether we need to provide a deprecated note for this protocol? If yes, then we need to call both old and new delegates from https://github.com/rakutentech/ios-miniapp/pull/163/files#diff-eaffa51bae0dd77ddb253a0469ff68a7914e6f8ff719a16a2638b706f0d64059L65. What do you think ?

Reply:
Updated